Core/src/Detector/CylindricalDetectorHelper.cpp

```diff
@@ -0,0 +1,1115 @@
+               boundValues[Acts::RectangleBounds::BoundValues::eMinX]);
+    // New joint bounds
+    bounds = std::make_unique<Acts::RectangleBounds>(
+        halfX, 0.5 * (boundaries[boundaries.size() - 1u] - boundaries[0u]));
+  } else if (binning == Acts::binZ) {
+    // Center R calculation, using projection onto vector
+    const auto& surfaceCenter = refSurface.center(gctx);
+    Acts::Vector3 centerDiffs = (surfaceCenter - volumeCenter);
+    Acts::ActsScalar centerR = centerDiffs.dot(refRotation.col(2));
+    // New joint center
+    Acts::Vector3 pCenter = volumeCenter + centerR * refRotation.col(2);
+    transform.pretranslate(pCenter);
+    // New joint bounds
+    Acts::ActsScalar halfY =
+        0.5 * (boundValues[Acts::RectangleBounds::BoundValues::eMaxY] -
+               boundValues[Acts::RectangleBounds::BoundValues::eMinY]);
+    bounds = std::make_unique<Acts::RectangleBounds>(
+        0.5 * (boundaries[boundaries.size() - 1u] - boundaries[0u]), halfY);
+  }
+  // Set the rotation
+  transform.prerotate(refRotation);
+  // A new surface on the negative side over the full range
+  auto surface = Acts::Surface::makeShared<Acts::PlaneSurface>(
+      transform, std::move(bounds));
+  // A make a portal and indicate the new link direction
+  Acts::Experimental::PortalReplacement pRep = {
+      Acts::Experimental::Portal::makeShared(surface), index, nDir, boundaries,
+      binning};
+  return pRep;
+}
+
+/// @brief  Helper method to strip side volumes from containers
+///
+/// @param containers the list of container
+/// @param sides the sides
+/// @param selectedOnly the selection restriction
+///
+/// @return a map of stripped out container
+std::map<unsigned int,
+         std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>>>
+stripSideVolumes(
+    const std::vector<Acts::Experimental::ProtoContainer>& containers,
+    const std::vector<unsigned int>& sides,
+    const std::vector<unsigned int>& selectedOnly = {},
+    Acts::Logging::Level logLevel = Acts::Logging::INFO) {
+  ACTS_LOCAL_LOGGER(Acts::getDefaultLogger("::stripSideVolumes", logLevel));
+
+  // Thes are the aripped out side volumes
+  std::map<unsigned int,
+           std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>>>
+      sideVolumes;
+
+  // Principle sides and selected sides, make an intersection
+  std::vector<unsigned int> selectedSides;
+  if (not selectedOnly.empty()) {
+    std::set_intersection(sides.begin(), sides.end(), selectedOnly.begin(),
+                          selectedOnly.end(),
+                          std::back_inserter(selectedSides));
+  } else {
+    selectedSides = sides;
+  }
+
+  for (const auto& pc : containers) {
+    for (const auto& s : selectedSides) {
+      if (pc.find(s) != pc.end()) {
+        auto p = pc.find(s)->second;
+        if (sideVolumes.find(s) == sideVolumes.end()) {
+          sideVolumes[s] = {};
+        }
+        auto& sVolumes = sideVolumes[s];
```

Comment:
```suggestion
        auto& sVolumes = sideVolumes[s];
```
I think this is the same because the map will construct an empty object if the key does not exist yet

Reply:
Ok. Updated.